### PR TITLE
update xterm.js dependency

### PR DESCRIPTION
**Pull request by athornton**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/12973

(addresses xterm.js bug fixed in 4.19.0: https://github.com/xtermjs/xterm.js/issues/3700)

Needs backport to 3.4.x as well.

## Code changes

Use 4.19.0 of xterm.js

## User-facing changes

Hopefully, emacs in a terminal will work again.

### CODE REVIEW

**Comment by fcollonval** [comment]
The update of xterm.js to 4.18.0 was done in https://github.com/jupyterlab/jupyterlab/pull/12715

**Comment by fcollonval** [comment]
bot please update galata snapshots

**Comment by fcollonval** [comment]
@athornton I opened https://github.com/lsst-sqre/jupyterlab/pull/1 against your PR that should fix the CI.

**Comment by athornton** [comment]
bot please update galata snapshot

**Comment by fcollonval** [comment]
@athornton the bot will not update your PR because of some rights limitation on GitHub (aka unable to commit on a PR branch from a fork in another organization). But you can still get the updated snapshots and commit them manually (what I did in https://github.com/lsst-sqre/jupyterlab/pull/1) by looking to the action https://github.com/jupyterlab/jupyterlab/actions/workflows/galata-update.yml

> Your are missing a **s** for executing the update command _please update galata snapshots_

**Comment by athornton** [comment]
bot please update galata snapshots

**Comment by athornton** [comment]
It's not wrong about the Check Release failure: it is indeed trying to look for documentation pages that are not there.  Don't know what to do about that.

**Comment by athornton** [comment]
@fcollonval is the doc check failure something I need to address, or does it just reflect changes in structure between released 3.4.x and unreleased 4.0.x?

File: package.json [hunk 1]
```diff
@@ -111,6 +111,7 @@
     "eslint-plugin-jest": "~26.5.3",
     "eslint-plugin-prettier": "~4.0.0",
     "eslint-plugin-react": "~7.29.4",
+    "hatchling": "^0.2.2",
```
Comment (fcollonval) [inline package.json:114]: @athornton Why is this additional dependency needed?

**Comment by fcollonval** [comment]
> is the doc check failure something I need to address, or does it just reflect changes in structure between released 3.4.x and unreleased 4.0.x?

You can ignore that failure it is indeed not related.

File: package.json (hunk 1 above)
Comment (athornton) [reply package.json:114]: Good question.  Let's take it out and see if it still works.  Probably a result of trying to run test suites locally and getting confused?

Comment (fcollonval) [inline package.json:114]:
```suggestion
```

Comment (fcollonval) [reply package.json:114]: @athornton you will have to apply this suggestion yourself as I don't have rights to apply them myself.

**Comment by athornton** [comment]
I think we're good, then?

File: package.json (hunk 1 above)
Comment (athornton) [reply package.json]: Done!

**Comment by athornton** [comment]
bot please update galata snapshots

**Comment by athornton** [comment]
@fcollonval Does the current state look reasonable now?

**Review by fcollonval — APPROVED** [review]
Thanks @athornton

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.4.x

**Comment by athornton** [comment]
Weird.  I get 

```
git cherry-pick -x -m1 d46c965a98217177e043ae90940f81348eda9d58
fatal: bad object d46c965a98217177e043ae90940f81348eda9d58
```

....but I can manually do the backport, anyway.

**Comment by fcollonval** [comment]
Hey @athornton, my bad I forget to merge before asking the bot to auto backport - hence the merge commit does not exist yet.

Would you mind resolving the conflict so I can merge this PR - you need to remove `@types/codemirror` from the `yarn.lock` file.

**Comment by athornton** [comment]
I've removed @types/codemirror and @types/tern from yarn.lock.

**Comment by fcollonval** [comment]
CI failure is not relevant

**Comment by thuzhf** [comment]
When will the next version released?

**Comment by fcollonval** [comment]
> When will the next version released?

I'm waiting for another issue to be fixed before doing a patch release. So expect a release within a week.

**Pull request merged**